Sources/PointFree/Auth.swift

```diff
@@ -5,14 +5,47 @@ import HttpPipeline
 import HttpPipelineHtmlSupport
 import Optics
 import Prelude
+import UrlFormEncoding
 
 let secretHomeResponse: (Conn<StatusLineOpen, Prelude.Unit>) -> IO<Conn<ResponseEnded, Data>> =
   writeStatus(.ok)
     >-> readGitHubSessionCookieMiddleware
     >-> respond(secretHomeView)
 
 let githubCallbackResponse =
-  authTokenMiddleware
+  extractGitHubAuthCode
+    <| authTokenMiddleware
```

Comment:
this is how we alter the `githubCallbackResponse` middleware to handle the optional code. we use a middleware transformer to convert the optional `String` to a non-optional, and in the case of `nil` we render a 400 bad request page.